src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -257,16 +257,22 @@
    {:db (update-in db [:wallet :ui :send] dissoc :token :token-display-name :tx-type)}))
 
 (rf/reg-event-fx :wallet/clean-selected-collectible
- (fn [{:keys [db]}]
-   (let [transaction-type (get-in db [:wallet :ui :send :tx-type])]
-     {:db (update-in db
-                     [:wallet :ui :send]
-                     dissoc
-                     :collectible
-                     :token-display-name
-                     :amount
-                     (when (send-utils/tx-type-collectible? transaction-type)
-                       :tx-type))})))
+ (fn [{:keys [db]} [{:keys [ignore-entry-point?]}]]
+   (let [entry-point-wallet-home? (= (get-in db [:wallet :ui :send :entry-point]) :wallet-stack)
+         multiple-owners?         (get-in db [:wallet :ui :send :collectible-multiple-owners?])
+         transaction-type         (get-in db [:wallet :ui :send :tx-type])]
+     (when (or ignore-entry-point?
+               (and entry-point-wallet-home? (not multiple-owners?))
+               (not entry-point-wallet-home?))
+       {:db (update-in db
+                       [:wallet :ui :send]
+                       dissoc
+                       :collectible
+                       :collectible-multiple-owners?
+                       :token-display-name
+                       :amount
+                       (when (send-utils/tx-type-collectible? transaction-type)
+                         :tx-type))}))))
```

Comment:
We skip cleaning the selected collectible on navigating back (the user will land on the `From` screen) from the select recipient screen if the collectible is present in multiple accounts.